src/NaCl.Core/Base/SalsaCore/Salsa20CoreIntrinsics.cs

```diff
@@ -0,0 +1,57 @@
+﻿#if INTRINSICS
```

Comment:
I wonder if we do need the preprocessor directive since we are checking if the hardware supports it